### PR TITLE
Heterodyne hotfixes to work with UHFQC v5

### PR DESCRIPTION
**Pull request by AdriaanRol**
@antsr , You may want to take a look at this as we made some changes to your recently merged enhancements to the HS driver. Everything should still be there but it would be good to test it on your side. 

Notable changes are 
- HS probe returns values in V now (unit aware plotting #131 and monitor #147 require SI to work) 
- Removed the uploading of the pulse to the UHFQC from the HS.prepare, after a discussion with @NielsBultink  we decided this does not belong here. 

This was tested to work on my setup before checking out the files I have and moving them to a branch for a PR.

### CODE REVIEW

File: pycqed/instrument_drivers/meta_instrument/heterodyne.py [hunk 1]
```diff
@@ -111,12 +112,21 @@ def common_init(self, name, LO, AWG, acquisition_instr='CBox',
...
-
+        self._eps = 0.01  # Hz slack for comparing frequencies
+        self._acq_marker_channels = ('ch4_marker1,ch4_marker2,' +
+                                     'ch3_marker1,ch3_marker2')
 
     def prepare(self, get_t_base=True):
+        # Uploading the AWG sequence
+        if (self._awg_seq_filename not in self.AWG.setup_filename() or
+                self._awg_seq_parameters_changed) and self.auto_seq_loading():
+            self._awg_seq_filename = \
```
Comment (AdriaanRol) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: @antsr I agree with that they should belong in the instrument specific prepare function. This was mostly my (quite brute) attempt at merging recent changes by Niels into yours until I got something that worked. It seems I missed some things there.

File: pycqed/instrument_drivers/meta_instrument/heterodyne.py [hunk 2]
```diff
@@ -160,40 +170,23 @@ def prepare_CBox(self, get_t_base=True):
...
-                    st_seqs.generate_and_upload_marker_sequence(
-                        5e-9, self.trigger_separation(), RF_mod=False,
-                        acq_marker_channels=self.acq_marker_channels())
-                self._awg_seq_parameters_changed = False
-
-        if self._UHFQC_awg_parameters_changed and self.auto_seq_loading():
-            self._acquisition_instr.awg_sequence_acquisition()
-            self._UHFQC_awg_parameters_changed = False
-
-        # prepare weights and rotation
```
Comment (AdriaanRol) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: I recall a discussion with @NielsBultink that led me to settle on this version of the prepare statement. I recall it has something to do with #135  and #94 . However I cannot recall the exact argument and I would argue you are right.

**Comment by AdriaanRol** [comment]
@antsr @johu2 I still have to add the most recent updates to the drivers by @NielsBultink as we've discussed in last Fridays telco. I'll add them to this PR (as this seems the right place for them).

**Comment by AdriaanRol** [comment]
@antsr @johu2 I've merged in the recent changes and updates from @NielsBultink into the current PR. This should contain the most up to date version of the UHFQC detectors. 

I've also made some changes to the heterodyne instrument. 
- I've removed the double AWG sequence loading as mentioned by @antsr 
- I've left the awg prepare statement where it is, this is the sequence that (if not using an LO modulated pulse) just plays the triggers and as such should remain where it is. (until we update it with a new style AWG sequence but I see no need). 
- I have added back in the UFHQC weight function prepare statements, they should indeed just be here (@NielsBultink let me know if you disagree). 

If everyone is OK I would like to merge this.

File: pycqed/instrument_drivers/meta_instrument/heterodyne.py [hunk 3]
```diff
@@ -522,48 +501,6 @@ def prepare_CBox(self, get_t_base=True):
...
-            int(np.log2(self.nr_averages())))
-        self._acquisition_instr.quex_wint_length(int(self.RO_length()*1.8e9))
-        # The AWG program uses userregs/0 to define the number of
-        # iterations in the loop
-        self._acquisition_instr.awgs_0_userregs_0(int(self.nr_averages()))
-        self._acquisition_instr.awgs_0_userregs_1(0) # 0 for rl, 1 for iavg
-        self._acquisition_instr.awgs_0_userregs_2(
-            int(self.acquisition_delay()*1.8e9/8))
-        self._acquisition_instr.awgs_0_single(1)
-
```
Comment (antsr) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: This should not be removed from the code as we still use it.

Comment (NielsBultink) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: Hi @antsr , Is the deletion you are asking about about the entire UHFQC prepare function or about some part of it? If it is about the entire function, it still exists in this file but in a slightly different place (it was defined twice)

Comment (antsr) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: I am asking about the UHFQC prepare function for the LO_modulated_heterodyne class, which differs from the UHFQC prepare for normal HeterodyneInstrument.

Comment (NielsBultink) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: Ah, OK, I have no idea how this happened. I have reinserted it.

File: pycqed/instrument_drivers/physical_instruments/ZurichInstruments/UHFQuantumController.py [hunk 4]
```diff
@@ -546,7 +590,9 @@ def prepare_DSB_weight_and_rotation(self, IF, weight_function_I=0, weight_functi
         cosI = np.array(np.cos(2*np.pi*IF*tbase))
         sinI = np.array(np.sin(2*np.pi*IF*tbase))
         eval('self.quex_wint_weights_{}_real(np.array(cosI))'.format(weight_function_I))
+        eval('self.quex_wint_weights_{}_real(np.array(sinI))'.format(weight_function_I))
         eval('self.quex_wint_weights_{}_real(np.array(sinI))'.format(weight_function_Q))
+        eval('self.quex_wint_weights_{}_real(np.array(cosI))'.format(weight_function_Q))
```
Comment (NielsBultink) [inline pycqed/instrument_drivers/physical_instruments/ZurichInstruments/UHFQuantumController.py]: I agree. anbd remove the two lines

File: pycqed/instrument_drivers/meta_instrument/heterodyne.py (hunk 2 above)
Comment (NielsBultink) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: I have put it back. Whenever heterodyne is used, both the integration functions and the AWG sequence should be overwritten. When going back to multiplexed readout afterwards, its calibration should be rerun.

**Comment by AdriaanRol** [comment]
@antsr , @NielsBultink updated the driver. Could you let us know if this is OK to merge?

File: pycqed/instrument_drivers/meta_instrument/heterodyne.py (hunk 1 above)
Comment (antsr) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: This change breaks `LO_modulated_Heterodyne.prepare` function that currently inherits the base `prepare` function from  `Heterodyne` by programming the Tektronix with a wrong sequence. I suggest either overriding `LO_modulated_Heterodyne.prepare` or moving the Tektronix AWG programming back to instrument specific prepare functions.

File: pycqed/instrument_drivers/meta_instrument/heterodyne.py [hunk 5]
```diff
@@ -525,8 +508,7 @@ def prepare_CBox(self, get_t_base=True):
     def prepare_UHFQC(self):
         """
         uses the UHFQC to generate the modulating signal and readout
-        """
-        # only uploads a seq to AWG if something changed
+        """# only uploads a seq to AWG if something changed
```
Comment (antsr) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: A newline went missing.

**Comment by AdriaanRol** [comment]
@NielsBultink , can you finish this?

File: pycqed/instrument_drivers/meta_instrument/heterodyne.py (hunk 5 above)
Comment (NielsBultink) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: fixed this

File: pycqed/instrument_drivers/meta_instrument/heterodyne.py (hunk 1 above)
Comment (NielsBultink) [inline pycqed/instrument_drivers/meta_instrument/heterodyne.py]: Have added prepare function without AWG uploading to the LO-modulated readout.

**Comment by NielsBultink** [comment]
I have pushed changes according to comment. I have however not been able to test LO modulated heterodyne as this is not implemented in my current setupas discussed with @AdriaanRol . @antsr   Could you please give OK if tested. Then we can merge with master.

**Pull request merged**